dashboard/index.js

```diff
@@ -1,23 +1,38 @@
 // launch the dashboard broker as a daemon
 const cp = require('child_process');
 const utils = require('./lib/utils/index');
+const mocks = require('./lib/mocks/index');
 
-const launchBroker = (port) => {
+const launchBroker = (port, isTest = false) => {
     if (!utils.validPortNumber(port)) {
         throw new Error('Invalid port number.');
     }
 
-    // spawn the broker as a child process
-    const brokerProcess = cp.spawn(
-        process.execPath,
-        ['./dashboard/server/index.js'],
-        {
-            detached: true, // runs the child process independently of parent
-            stdio: ['ignore', 'ignore', 'ignore', 'ipc'],
-            // disconnect all pipes except for ipc
-            // ['stdin', 'stdout', 'stderr', 'ipc']
-        }
-    );
+    let brokerProcess;
+
+    if (!isTest) {
+        // spawn the broker as a child process
+        brokerProcess = cp.spawn(
+            process.execPath,
+            ['./dashboard/server/index.js'],
+            {
+                detached: true,
+                // runs the child process independently of parent
+                stdio: ['ignore', 'ignore', 'ignore', 'ipc'],
+                // disconnect all pipes except for ipc
+                // ['stdin', 'stdout', 'stderr', 'ipc']
+            }
+        );
+    } else {
+        brokerProcess = mocks.mockSpawn(
```

Comment:
This is not how we use mocking. You should stub the function that is external in the test itself. The source should never know about the tests.

Ref:
1. Mocking in newman: https://github.com/postmanlabs/newman/blob/develop/test/unit/cli.test.js
2. https://sinonjs.org/

Reply:
@coditva Thank you for this! I have implemented sinonjs for stubbing at https://github.com/elit-altum/newman-dashboard/pull/7 could we move this discussion there?